File: backend/src/main/java/com/tododuk/domain/user/dto/UserDto.java
package com.tododuk.domain.user.dto;

import com.tododuk.domain.user.entity.User;

import java.time.LocalDateTime;

public record UserDto(
    int id,
    String nickname,
    String email,
    String profileImageUrl,
    LocalDateTime createDate,
    LocalDateTime modifyDate
){
    public UserDto(User user) {
        this(
            user.getId(),
            user.getNickName(),
            user.getUserEmail(),
            user.getProfileImgUrl(),
            user.getCreateDate(),
            user.getModifyDate()
        );
    }
}

File: backend/src/main/java/com/tododuk/domain/user/dto/UserDto.kt
package com.tododuk.domain.user.dto

import com.tododuk.domain.user.entity.User
import java.time.LocalDateTime

@JvmRecord
data class UserDto(
    val id: Int,
    @JvmField val nickname: String,
    val email: String,
    @JvmField val profileImageUrl: String,
    val createDate: LocalDateTime,
    val modifyDate: LocalDateTime
) {
    constructor(user: User) : this(
        user.id,
        user.nickName,
        user.userEmail,
        user.profileImgUrl,
        user.createDate,
        user.modifyDate
    )
}



